### PR TITLE
Fix memleak issue of #1050 

### PR DESCRIPTION
**Pull request by JaylinYu**
related to https://github.com/emqx/nanomq/issues/1043

### CODE REVIEW

**Review by alvin1221 — APPROVED** [review]

File: src/sp/protocol/mqtt/mqtt_parser.c [hunk 1]
```diff
@@ -672,35 +673,37 @@ conn_handler(uint8_t *packet, conn_param *cparam, size_t max)
...
-			    cparam->payload_format_indicator == 0x01) {
-				cparam->will_msg.body =
-				    (char *) copyn_utf8_str(
-				        packet, &pos, &len_of_str, max - pos);
-			}
-			cparam->will_msg.len = len_of_str;
-			rv = len_of_str <= 0 ? PAYLOAD_FORMAT_INVALID : 0;
-			log_trace(
-			    "will_msg: %s %d", cparam->will_msg.body, rv);
+		if (rv == 0 && cparam->payload_format_indicator == 0) {
```
Comment (wanghaEMQ) [inline src/sp/protocol/mqtt/mqtt_parser.c:692]: rv has been checked at line 687. The check for rv here is unnecessary

File: src/sp/protocol/mqtt/mqtt_parser.c [hunk 2]
```diff
@@ -672,35 +673,37 @@ conn_handler(uint8_t *packet, conn_param *cparam, size_t max)
...
+			cparam->will_msg.body = (char *) copyn_str(
+			    packet, &pos, &len_of_str, max - pos);
+		} else if (rv == 0 &&
```
Comment (wanghaEMQ) [inline src/sp/protocol/mqtt/mqtt_parser.c:695]: Unnecessary check for rv.

File: src/sp/protocol/mqtt/nmq_mqtt.c [hunk 3]
```diff
@@ -995,6 +1000,7 @@ nano_pipe_recv_cb(void *arg)
 			nni_pipe_close(p->pipe);
 			return;
 		}
+		conn_param_clone(cparam);
```
Comment (wanghaEMQ) [inline src/sp/protocol/mqtt/nmq_mqtt.c:1003]: Clone must before being used!!!

File: src/sp/protocol/mqtt/nmq_mqtt.c [hunk 4]
```diff
@@ -1020,6 +1025,7 @@ nano_pipe_recv_cb(void *arg)
...
```
Comment (wanghaEMQ) [inline src/sp/protocol/mqtt/nmq_mqtt.c:1028]: CLone must before being used!!!

**Review by wanghaEMQ — CHANGES_REQUESTED** [review]
Hi. Plz review the comments. There are some critical changes

**Review by lee-emqx — APPROVED** [review]

**Review by wanghaEMQ — APPROVED** [review]

**Pull request merged**